js/impact.js

```diff
@@ -0,0 +1,2332 @@
+         if (currentCompounds.hasOwnProperty(compoundID)) {
+            // If the compound is still here in the current state
+            var currentCompound = currentCompounds[compoundID];
+
+            // Check for updates ...
+            if (compound.name != currentCompound.name
+               || compound.color != currentCompound.color) {
+               compoundsDelta[compoundID] = {
+                  action: DELTA_ACTION_UPDATE,
+                  name  : currentCompound.name,
+                  color : currentCompound.color
+               };
+            }
+
+            // Remove it from the currentCompounds data
+            delete currentCompounds[compoundID];
+         } else {
+            // If the compound is missing in the current state, it's been deleted
+            compoundsDelta[compoundID] = {
+               action           : DELTA_ACTION_DELETE,
+            };
+         }
+      });
+
+      // Now iterate on the compounds we have in the current state
+      Object.keys(currentCompounds).forEach(function (compoundID) {
+         compoundsDelta[compoundID] = {
+            action: DELTA_ACTION_ADD,
+            name  : currentCompounds[compoundID].name,
+            color : currentCompounds[compoundID].color
+         };
+      });
+
+      return compoundsDelta;
+   },
+
+   /**
+    * Delta computation for parents
+    *
+    * @returns {Object}
+    */
+   computeItemsDelta: function(currentNodes) {
+      var itemsDelta = {};
+
+      // Prepare position map
+      var nodesPositions = {};
+      impact.cy.filter("node:childless").forEach(function(node) {
+         nodesPositions[node.data('id')] = node.position();
+      });
+
+      // Now iterate on the parents we have in the current state
+      Object.keys(currentNodes).forEach(function (nodeID) {
+         var node = currentNodes[nodeID];
+         itemsDelta[node.impactitem_id] = {
+            action   : DELTA_ACTION_UPDATE,
+            parent_id: node.parent,
+         };
+
+         // Set parent to 0 if null
+         if (node.parent == undefined) {
+            node.parent = 0;
+         }
+
+         if (nodeID == impact.startNode) {
+            // Starting node of the graph, save viewport and edge colors
+            itemsDelta[node.impactitem_id] = {
+               action                  : DELTA_ACTION_UPDATE,
+               parent_id               : node.parent,
+               position_x              : node.position.x,
+               position_y              : node.position.y,
+               zoom                    : impact.cy.zoom(),
+               pan_x                   : impact.cy.pan().x,
+               pan_y                   : impact.cy.pan().y,
+               impact_color            : impact.edgeColors[FORWARD],
+               depends_color           : impact.edgeColors[BACKWARD],
+               impact_and_depends_color: impact.edgeColors[BOTH],
+               nodes_positions         : nodesPositions,
```

Comment:
I cannot save impact graph unless I made this change:
```suggestion
               nodes_positions         : JSON.stringify(nodesPositions),
```